pkg/codestruct/codestruct.go

```diff
@@ -0,0 +1,39 @@
+package codestruct
+
+import (
+	"go/ast"
+)
+
+type Weights struct {
+	Func      uint `xml:"func"`
+	Struct    uint `xml:"struct"`
+	Interface uint `xml:"interface"`
+}
+
+type Metric struct {
+	Config Weights
+	Comp   uint
+}
+
+func (m *Metric) ParseNode(n ast.Node) {
+	m.Comp += getCodeStructComp(n, &m.Config)
+}
+
+func getCodeStructComp(n ast.Node, cfg *Weights) uint {
+	comp := uint(0)
+
+	switch n.(type) {
+	case *ast.StructType:
+		comp += cfg.Struct
```

Comment:
Why use += (weight) here when the result is always 0 + (weight) = (weight)? It seems that just assigning the weight is more readable

Reply:
My bad, I'll fix it